src/main/java/duke/Duke.java

```diff
@@ -43,49 +40,43 @@ public void run() throws IOException {
 
                 switch (command.getCommandType()) {
                     case LIST_CMD:
-                        ui.printLine("Here are your task(s):");
-
-                        int len = taskList.getNumTasks();
-
-                        for (int i = 0; i < len; ++i) {
-                            t = taskList.getTask(i);
-                            ui.printLine("    " + (i + 1) + ". " + t);
-                        }
-                        ui.printLine();
+                        ui.printTaskList(taskList);
                         break;
                     case DONE_CMD:
-                        index = Integer.parseInt(command.getParams()[0]) - 1;
+                        index = extractTaskIndexFromCmdParam(command);
 
                         t = taskList.getTask(index);
                         taskList.markAsDone(index);
-                        ui.printLine("Nice! I've marked this task as done:" + Ui.LF + "    " + t + Ui.LF);
+                        ui.printTaskMarkedDone(t);
+
                         storage.saveTasksToFile(taskList);
                         break;
                     case DELETE_CMD:
-                        index = Integer.parseInt(command.getParams()[0]) - 1;
+                        index = extractTaskIndexFromCmdParam(command);
 
                         t = taskList.getTask(index);
                         taskList.deleteTask(index);
 
-                        ui.printLine("Noted! I've removed this task:" + Ui.LF + "    " + t + Ui.LF
-                                + "Now, you have " + taskList.getNumTasks() + " item(s) in your list." + Ui.LF);
+                        ui.printTaskDeleted(t, taskList);
 
                         storage.saveTasksToFile(taskList);
-
                         break;
                     case TODO_CMD:
                         t = new Todo(command.getParams()[0]);
                         addAndPrintTask(t, taskList);
+
                         storage.saveTasksToFile(taskList);
                         break;
                     case DEADLINE_CMD:
                         t = new Deadline(command.getParams()[0], LocalDate.parse(command.getParams()[1]));
                         addAndPrintTask(t, taskList);
+
                         storage.saveTasksToFile(taskList);
                         break;
                     case EVENT_CMD:
                         t = new Event(command.getParams()[0], LocalDate.parse(command.getParams()[1]));
                         addAndPrintTask(t, taskList);
+
```

Comment:
In this module, when you use the switch, try to do it like this:
switch (command.getCommandType()) {
case EVENT_CMD: 
...
}